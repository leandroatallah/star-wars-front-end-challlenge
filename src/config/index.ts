export const CHANGE_THEME = 'CHANGE_THEME'
export const THEME_DARK_SIDE = 'THEME_DARK_SIDE'
export const THEME_LIGHT_SIDE = 'THEME_LIGHT_SIDE'
