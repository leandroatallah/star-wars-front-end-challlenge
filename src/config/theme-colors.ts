type myTheme = {
    colors: {
        blue: string
        yellow: string
        black: string
        white: string
    }
}

const themeColors: myTheme = {
    colors: {
        blue: '#2C97D1',
        yellow: '#FBFE63',
        black: '#2A2A2A',
        white: '#FFF',
    },
}

export default themeColors