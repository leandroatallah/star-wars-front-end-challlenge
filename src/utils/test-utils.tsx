import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { CHANGE_THEME } from '../config'
import { createStore, Store } from 'redux'

const startingState = { theme: '' }
function reducer(state = startingState, action: { type: string; payload: string }) {
    switch (action.type) {
        case CHANGE_THEME:
            return { ...state, theme: action.payload }
        default:
            return state
    }
}

type storeProps = {
    initialState?: {
        theme: string
    },
    store?: Store<{
        theme: string;
    }, {
        type: string;
        payload: string;
    }>
}

export function renderWithRedux(
    component: JSX.Element,
    { initialState, store = createStore(reducer, initialState) }: storeProps = {},
) {
    return {
        ...render(<Provider store={store} > {component} </Provider>),
    }
}
