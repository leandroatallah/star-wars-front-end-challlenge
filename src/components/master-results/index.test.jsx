import '@testing-library/jest-dom'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import MasterResults from '.'
import themeColors from '../../config/theme-colors'
import { THEME_DARK_SIDE, THEME_LIGHT_SIDE } from '../../config'
import { cleanup } from '@testing-library/react'
import { createStore } from 'redux'
import { renderWithRedux } from '../../utils/test-utils'

afterEach(cleanup)

describe('<MasterResults />', () => {
    it('Render with redux', () => {
        const { getByTestId } = renderWithRedux(
            <ThemeProvider theme={themeColors}>
                <MasterResults />
            </ThemeProvider>,
        )
        expect(getByTestId('button')).toMatchSnapshot()
    })

    it('should have Luke Skywalker as master when light theme is setted', () => {
        const store = createStore(() => ({ theme: THEME_LIGHT_SIDE }))
        const { getByRole } = renderWithRedux(
            <ThemeProvider theme={themeColors}>
                <MasterResults />
            </ThemeProvider>,
            { store },
        )
        expect(getByRole('heading')).toHaveTextContent('Luke Skywalker')
    })

    it('should have Darth Vader as master when dark theme is setted', () => {
        const store = createStore(() => ({ theme: THEME_DARK_SIDE }))
        const { getByRole } = renderWithRedux(
            <ThemeProvider theme={themeColors}>
                <MasterResults />
            </ThemeProvider>,
            { store },
        )
        expect(getByRole('heading')).toHaveTextContent('Darth Vader')
    })
})
