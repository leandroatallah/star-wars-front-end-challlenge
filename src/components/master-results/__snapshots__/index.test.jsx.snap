// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MasterResults /> Render with redux 1`] = `
.c0 {
  background-color: #2C97D1;
  display: inline-block;
  height: 56px;
  font-size: 16px;
  line-height: 1.218;
  font-weight: 700;
  padding: 0 37px;
  border-radius: 10px;
  color: #FFF;
  cursor: pointer;
}

<button
  class="c0"
  data-testid="button"
  type="button"
>
  choose your path again, Padawan
</button>
`;
